cmd/ipfs/kubo/daemon.go

```diff
@@ -458,6 +458,7 @@ func daemonFunc(req *cmds.Request, re cmds.ResponseEmitter, env cmds.Environment
 			cfg.Identity.PeerID,
 			cfg.Addresses,
 			cfg.Identity.PrivKey,
+			cfg.Experimental.HTTPRetrieval.Enabled,
```

Comment:
nit: using `Experimental.*` turned out  to be a headache – requires migration or special code that guides users to switch to new value once we move it (like we did with `AcceleratedDHTClient` that now lives in `Routing.*`). 

I did not use it for `AutoTLS` config, i've put it in the top level from the start, and just marked it as **EXPERIMENTAL** when documenting in `/docs/config.md` 

Maybe do the same here? Have top-level `HTTPRetrieval` mark as **EXPERIMENTAL**.